### PR TITLE
Solved issue "Enhancement to MEWS Score"

### LINKED ISSUE #7373: Enhancement to MEWS Score
**Is your feature request related to a problem? Please describe.**
Currently, for MEWS Score to be calculated all 5 data points need to be entered in the patient records through a single log update.
The 5 data points required to calculate MEWS are:
MEWS

All the above parameters other than Consciousness level, are auto-updated in CARE through BPL 5-para integration through virtual nursing assistant. if the nurse is allowed to update only consciousness level additionally, MEWS can be calculated. However, with the way the feature is developed right now, because all data points needs to be entered in a single log update, this is not possible. 

**Describe the solution you'd like**

- [ ] Allow calculation of MEWS score by fetching the relevant 5 data points spread across multiple log updates filed within a time frame of 30 min. 

Eg: 
Resp rate - updated at 10AM
Heart Rate- updated at 10AM
Systolic BP- updated at 10:15AM
Temp- updated at 10:15AM
Consciousness- updated at 10:29AM

In this case- MEWS will be calculated based on the 5 parameters, since they are all updated within a time frame of 30 min (10-10:29AM)

Subsequently if
Resp rate - revised at 10:35 AM
Heart Rate- updated at 10:35 AM

MEWS will be revised based on the entries made between 10:15AM to 10:45AM since 2/5 parameters are revised, and the 5 parameters are entered in the time frame of 30 min (10:15AM to 10:45AM)

However, later on if Systolic BP is revised again at 11:45AM, MEWS will not be revised since, only 1 parameters is updated and remaining 4 parametrers are not available within 30 min bracket.

- [ ] Add measured at time under the MEWS on Patient dashboard. mention the time stamp of the latest of the 5-parameters.

### PR DESCRIPTION
**Pull request by r-nikhilkumar**
- Fixes #7373 
- Allow calculation of MEWS score by fetching the relevant 5 data points spread across multiple log updates filed within a time frame of 30 min✅
- Add measured at time under the MEWS on Patient dashboard. mention the time stamp of the latest of the 5-parameters✅

depends on https://github.com/coronasafe/care/pull/1991

@coronasafe/care-fe-code-reviewers @coronasafe/code-reviewers

### CODE REVIEW

**Comment by r-nikhilkumar** [comment]
@rithviknishad, please remove stale label

**Comment by nihal467** [comment]
@r-nikhilkumar 
image

the deploy build is failing fix it

**Comment by r-nikhilkumar** [comment]
@nihal467 first need to merge the backend then the test will be passed

**Comment by rash-27** [comment]
@r-nikhilkumar ig the problem is with the typescript errors , see the screenshot attached

**Comment by r-nikhilkumar** [comment]
@rash-27 yeah, sorry I didn't see ss

**Comment by rithviknishad** [comment]
@r-nikhilkumar We've recently introduced type-checks at the build step to voluntarily fail the build if there were type errors. This was done to improve code quality and catch potential bugs early on before it even gets merged. Ref: https://github.com/coronasafe/care_fe/pull/7469

It'd be appreciated if you could fix the type errors introduced in this PR.

**Comment by r-nikhilkumar** [comment]
@nihal467 solved the typeError now you can test it, sorry for inconvenience

**Review by rithviknishad — CHANGES_REQUESTED** [review]
Proper types needed. Setting to `any` defeats the purpose of enforcing type-safety.

**Comment by r-nikhilkumar** [comment]
so for that I need to make changes in consultationModel interface and I also need to create an interface for mewFields with all the required attribute, could I make changes in model interface?

File: src/Components/Facility/Consultations/Mews.tsx [hunk 1]
```diff
@@ -1,6 +1,6 @@
-import { DailyRoundsModel } from "../../Patient/models";
 import { formatDateTime } from "../../../Utils/utils";
-
+import { MewsFieldModel } from "../../Patient/models";
+// import { mewsFieldsModel } from "../../Patient/models";
```
Comment (rithviknishad) [inline src/Components/Facility/Consultations/Mews.tsx]:
```suggestion
import { MewsFieldModel } from "../../Patient/models";
```

**Review by rithviknishad — APPROVED** [review]

**Comment by nihal467** [comment]
image

**Above is PR deployment**

image

**Above is the current staging**

@r-nikhilkumar When there is a log update present for a patient, and the MEWS relevant data are missing, the MEWS score should be displayed as "-" Upon hovering over it, the related tooltip should show the missing data and related information. This functionality is currently working properly in the staging environment but is missing in your pull request.

It is a basic functionality of the MEWS score. Therefore, I kindly request you to familiarize yourself with the function first, understand its functionality, make the necessary changes, conduct self QA of the function, and then submit it for review.

**Comment by r-nikhilkumar** [comment]
@nihal467 it is because I made this change
image


in my local its showing the mews field as you shown in staging view, it is not showing in deployed preview might be a reason that from backend mews_field is not getting sent as a response field and to get mews_field in response first need to make changes from backend and for the backend changes i raised a pr https://github.com/coronasafe/care/pull/1991 already once that will be deployed in main server, then the mews_field response will be fetched successfully and the mews score will be shown too
image

**Comment by nihal467** [comment]
discussion is going on in the backend repo : https://github.com/coronasafe/care/pull/1991

**Comment by r-nikhilkumar** [comment]
@rithviknishad what about this pr, is this going to be wasted? no use?😞

**Comment by nihal467** [comment]
@r-nikhilkumar closing this PR and unassigning you from this issue due to inactivity, feel free to reopen the same if you are still working on it

**Comment by r-nikhilkumar** [comment]
@nihal467 is the issue still in progress and not resolved, why you reopened this my PR?🤔

**Comment by nihal467** [comment]
@r-nikhilkumar https://github.com/ohcnetwork/care_fe/issues/7373#issuecomment-2296147719 , can you post a reply to this message

**Comment by bodhish** [comment]
Closing this PR due to lack of recent progress.

**Pull request closed without merging**